### PR TITLE
Update documentation about removal of private keys 

### PR DESCRIPTION
**Pull request by JoaoSa-da**
Covers both these Canton issues: https://github.com/DACH-NY/canton/issues/22368 and https://github.com/DACH-NY/canton/issues/22147

### CODE REVIEW

File: docs/2.10.0/docs/canton/usermanual/security.rst [hunk 1]
```diff
@@ -320,6 +320,29 @@ A fingerprint of a key can be retrieved from the list of public keys:
...
+
+.. literalinclude:: /canton/includes/mirrored/enterprise/app/src/test/scala/com/digitalasset/canton/integration/tests/security/topology/KeyManagementIntegrationTest.scala
+   :language: scala
+   :start-after: user-manual-entry-begin: DeletePrivateKey
+   :end-before: user-manual-entry-end: DeletePrivateKey
+   :dedent:
+
+.. warning::
+  Exercise caution when deleting a private key. You must ensure that the key is no longer in use and is not needed
+  for decrypting or verifying signatures of previous messages. Therefore, keys must only be deleted after you have
```
Comment (andreaslochbihler-da) [inline docs/2.10.0/docs/canton/usermanual/security.rst:340]: A private key is not needed to verify signatures. You need the corresponding public key for verification and this section is about the private key.

File: docs/2.10.0/docs/canton/usermanual/security.rst [hunk 2]
```diff
@@ -320,6 +320,29 @@ A fingerprint of a key can be retrieved from the list of public keys:
...
+  pruned all data that directly or indirectly uses them. This is especially crucial for sequencers in open networks.
```
Comment (andreaslochbihler-da) [inline docs/2.10.0/docs/canton/usermanual/security.rst:341]: There is no guidance on how an operator can figure out whether all data has been pruned that directly or indirectly uses them. A conservative rule of thumb would be to say that you should only delete the key when the node was pruned for a timestamp that is later than the point in time when the key was deactivated.

File: docs/2.10.0/docs/canton/usermanual/security.rst [hunk 3]
```diff
@@ -320,6 +320,29 @@ A fingerprint of a key can be retrieved from the list of public keys:
...
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Deletion of keys, in particular, private keys is not done automatically in Canton. When a key is rotated, the previous
+key remains stored, but is disabled and no longer used. To permanently delete a key, use the following Canton console
+command:
+
+.. literalinclude:: /canton/includes/mirrored/enterprise/app/src/test/scala/com/digitalasset/canton/integration/tests/security/topology/KeyManagementIntegrationTest.scala
+   :language: scala
+   :start-after: user-manual-entry-begin: DeletePrivateKey
+   :end-before: user-manual-entry-end: DeletePrivateKey
```
Comment (andreaslochbihler-da) [inline docs/2.10.0/docs/canton/usermanual/security.rst:335]: Canton's main does not seem to have those delimiters yet. Can you point me to a PR that adds them?

Comment (JoaoSa-da) [reply docs/2.10.0/docs/canton/usermanual/security.rst:335]: https://github.com/DACH-NY/canton/pull/22374 and https://github.com/DACH-NY/canton/pull/22375

**Review by andreaslochbihler-da — APPROVED** [review]
LGTM. Thanks for documenting this.

File: docs/2.10.0/docs/canton/usermanual/security.rst [hunk 4]
```diff
@@ -320,6 +320,30 @@ A fingerprint of a key can be retrieved from the list of public keys:
...
```
[the remark this answers is not in the document]
Comment (JoaoSa-da) [reply docs/2.10.0/docs/canton/usermanual/security.rst:335]: I opened an issue so I don't forget it https://github.com/DACH-NY/canton/issues/22595

**Review by carrielaben-da — APPROVED** [review]

**Pull request merged**